frontend/src/metabase-lib/v1/queries/utils/column-key.ts

```diff
@@ -67,15 +74,23 @@ export const getColumnSettings = (
   return getObjectColumnSettings(settings?.column_settings, column);
 };
 
-// Gets the corresponding viz settings for the column. We check for both
-// legacy and modern keys because not all viz settings have been migrated.
-// To be extra safe and maintain backward compatibility, we check for the
-// legacy key first.
+/**
+ * Gets the corresponding viz settings for the column. We check for both
+ * legacy and modern keys because not all viz settings have been migrated.
+ * To be extra safe and maintain backward compatibility, we check for the
+ * legacy key first.
+ *
+ * Sometimes the `field_ref` has the `base-type`, sometimes it doesn't. To not
+ * break legacy viz settings, try to get the column settings with and without
+ * `base-type`.
+ */
 export const getObjectColumnSettings = (
   settings: Record<string, ColumnSettings> | null | undefined,
   column: DatasetColumnReference,
 ) => {
   return (
-    settings?.[getLegacyColumnKey(column)] ?? settings?.[getColumnKey(column)]
+    settings?.[getLegacyColumnKey(column)] ??
+    settings?.[getLegacyColumnKey(column, { excludeBaseType: true })] ??
```

Comment:
so this would still fail if the key has a base type but it's slightly different, like `type/BigInteger` versus `type/Integer`, right?

Reply:
I'll handle it separately, good point